server/neptune/gateway/event/preworkflow/hook_executor.go

```diff
@@ -0,0 +1,64 @@
+package preworkflow
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"os"
+	"os/exec"
+)
+
+// interface for actual execution makes local testing easier
+
+type HookExecutor interface {
```

Comment:
should we move this interface private and move it to the hooks_runnner.go?

Reply:
Not unless we keep the mock versions of the `HookExecutor` in the source files or make the test files share the source package name. I did move the interface to the hooks runner file though.

Reply:
Misunderstood this to mean making the un-exporting the interface's Execute function. Fixed it to make the interface itself private.